File: Java/src/main/java/com/gildedrose/items/SulfurasItem.java
package com.gildedrose.items;

import com.gildedrose.Item;

public class SulfurasItem extends GildedItem {
    public SulfurasItem(Item item) { super(item); }

    @Override
    public void updateQuality() {
        // Sulfuras ne change jamais
    }
}

File: php/src/Items/SulfurasItem.php
<?php

declare(strict_types=1);

namespace GildedRose\Items;

use GildedRose\Item;
use GildedRose\Items\Abstract\AbstractItem;
use GildedRose\Items\Interface\ItemInterface;

class SulfurasItem extends AbstractItem implements ItemInterface
{
    public function updateQuality(Item $item): void
    {
        // Do nothing, Sulfuras never changes in quality
    }
}
